### PR TITLE
Week 12: Movies

### PR DESCRIPTION
**Pull request by monkeybusiness101**
Link to my deployed project:

https://wizardly-goldwasser-9c77ac.netlify.app/

My reflections on how this week's project turned out:

It was fun to work with APIs, useState, useEffect and newly introduced react Router.

Things I'd like to have clarified or explained in more detail:

How to handle the second fetch of the backdrop the returns undefined. I saw Maks lecture on Friday so need to dig into conditionals a bit more

I also have not looked into bad cases where users ends up on a site that does not exist for example

### CODE REVIEW

File: code/public/index.html [hunk 1]
```diff
@@ -4,31 +4,15 @@
...
-      Only files inside the `public` folder can be referenced from the HTML.
-
-      Unlike "/favicon.ico" or "favicon.ico", "%PUBLIC_URL%/favicon.ico" will
-      work correctly both with client-side routing and a non-root public URL.
-      Learn how to configure a non-root public URL by running `npm run build`.
-    -->
-    <title>Technigo React App</title>
+    <link href="https://fonts.googleapis.com/css2?family=Roboto:wght@700&display=swap" rel="stylesheet">
+    
+    <title>Movies</title>
```
Comment (sandra-shumanteva-kaevska) [inline code/public/index.html:9]: Nice that you think of changing title :)

File: code/src/BackButton.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import React from 'react'
+
+export const BackButton = () => {
+    return (
+        <div className="back-button">
+        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 30 30">
+            <path d="M27 14.5C27 7.596441 21.4035594 2 14.5 2S2 7.596441 2 14.5 7.5964406 27 14.5 27 27 21.403559 27 14.5zm-19.3388348-.353553l7.4852814-7.485282c.1952622-.195262.5118446-.195262.7071068 0l2.1213203 2.121321c.1952622.195262.1952622.511844 0 .707106L12.9644661 14.5l5.0104076 5.010408c.1952622.195262.1952622.511844 0 .707106l-2.1213203 2.121321c-.1952622.195262-.5118446.195262-.7071068 0l-7.4852814-7.485282c-.19799-.19799-.197989-.509117 0-.707106z" fill="#fff" fillRule="evenodd"></path>
```
Comment (sandra-shumanteva-kaevska) [inline code/src/BackButton.js:7]: It is great that you have arrow button from "w3" 👍

File: code/src/index.css [hunk 3]
```diff
@@ -1,13 +1,135 @@
...
+
+.back-button {
+  transform: translate(100%, 91%);
+  width: 50px;
+  height: 50px;
+}
+
+/* mobile view */
+
+@media only screen and (max-width: 400px) {
```
Comment (sandra-shumanteva-kaevska) [inline code/src/index.css:122]: It is good that you think to make mobile responsive even it is not requirement 👍

**Review by sandra-shumanteva-kaevska — COMMENTED** [review]
Nice structured, easy readable code and great naming of components. The web page is working properly without any bugs 👍  Great job.

 I think that you should just delete "console.log" in your code.

File: code/src/Movies.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    
+    useEffect(() => {
+        
+        fetch('https://api.themoviedb.org/3/movie/popular?api_key=eb281b7f8ece67c3b577de3061fe96e1&language=en-US&page=1')
+        .then((response) => {
+            return response.json()
+        })
+        .then((text) => {
+            fetchMovies(text.results)
+            console.log(text.results)
```
[the remark this answers is not in the document]
Comment (EmOlsz) [reply code/src/Movies.js:16]: 👍

File: code/src/Movies.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+            console.error('Request failed', error)
+        })
+
+    }, [])
+
+    return <div className="movie-container">
+            {
+                movies.map(movie => (  
+                 <div className="movie">       
+                    <Link to={`${movie.id}`} key={movie.id}>
```
Comment (EmOlsz) [inline code/src/Movies.js:28]: `key` attribute should be added to root element inside loop (so, parent of all elements inside loop). In your case, it's `<div className="movie"> `, not `<Link to={`${movie.id}`} key={movie.id}>`

File: code/src/Movies.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (monkeybusiness101) [reply code/src/Movies.js:16]: fixed

File: code/src/ShowMovie.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+    useEffect(() => {
+
+    fetch(`https://api.themoviedb.org/3/movie/${id}?api_key=eb281b7f8ece67c3b577de3061fe96e1&language=en-US`)
+    .then((response) => {
+        return response.json()
+    })
+    .then((text) => {
+        getDetails(text)
+        console.log(text)
```
[the remark this answers is not in the document]
Comment (monkeybusiness101) [reply code/src/ShowMovie.js:21]: fixed

File: code/src/Movies.js (hunk 5 above)
Comment (monkeybusiness101) [reply code/src/Movies.js:28]: fixed

**Comment by monkeybusiness101** [comment]
Hi @sandra-shumanteva-kaevska and @EmOlsz Thank you for the input and feedback!

I have removed console.log, moved the key to the parent and removed some spacing. Can you please have a look and approve my PR?

/Max

**Pull request closed without merging**